packages/monorepo-builder/tests/ConflictingUpdater/Fixture/expected_composer.json

```diff
@@ -0,0 +1,8 @@
+{
+    "require": {
+        "symplify/package-builder": "^9.2"
+    },
+    "conflict": {
+        "symplify/another-package": "<9.2.0"
+    }
+}
```

Comment:
@enumag This is a text fixture file, that will be decorated with single conflicting package. The input file is right bellow. I need help with this section, as I have never used it before.

AFAIK the "conflict" section should contain all symplify packages, except the one in "reqire/require-dev/name".  Is that so?

But that would be ~30-35 "conflict" items on every `composer.json`, which seems a bit weird to me. How could we lower this number?

Reply:
what is your intention? disable all combination of `x.y` with `x.(y-1)`?

Reply:
My goal is to avoid the bugs you've reported. Would this help?

Reply:
> My goal is to avoid the bugs you've reported. 

Considering that you closed the main issue as "false positive", I don't quite believe this statement. 🙄 

Adding conflicts for all packages automatically is a bad idea. If you want to limit it that way then you can easily achieve the same restrictions with standard composer version expressions. Conflicts are only good if done manually when there is a reason to add one.

As for how to avoid the bugs, well adopt the standard semver good practices such as not changing API in a patch version.

Reply:
Your main problem which is causing issues in my opinion is what you consider "public API". You're still thinking of it as if Symplify was one large package which isn't the case. If a class is in one symplify package A and used in package B then it's no longer internal. It becomes part of public API of package A.

Reply:
We can only automate it, as I will definitelly forget it next time :D 

I assume the packages should be determined by inversed relationship.

E.g.

- package A requires B.
- new version is released
- B should have conflict with A of x-1

Reply:
> We can only automate it, as I will definitelly forget it next time :D

With Roave/BackwardCompatibilityCheck you wouldn't.

> I assume the packages should be determined by inversed relationship.

Yup. Package A requires package B, then package B has a BC break => package B should add a conflict with older package A.

In the most recent [case](https://github.com/symplify/symplify/issues/3075) symplify/astral 9.2.7+ should have a conflict with symplify/phpstan-rules <= 9.2.6.

Reply:
> With Roave/BackwardCompatibilityCheck you wouldn't.

I think implementation of that packages is impossible to use in monorepo. 

> Yup. Package A requires package B, then package B has a BC break => package B should add a conflict with older package A.

Thanks for confirmation :+1: